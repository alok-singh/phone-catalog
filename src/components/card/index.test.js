import React from "react";
import Card from "./index";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

const renderApp = (props) => {
  return render(
    <BrowserRouter>
      <Card {...props} />
    </BrowserRouter>
  );
};

beforeEach(() => {});

afterEach(() => {
  cleanup();
});

test("Card renders correctly", () => {
  const props = {
    title: "title",
    price: "price",
    image: "image",
    link: "link",
  };
  
  const { container } = renderApp(props);
  expect(container.querySelector("a").attributes.href.value).toBe(`/${props.link}`);
  expect(container.querySelector("img").attributes.src.value).toBe(props.image);
  expect(container.querySelector("img").attributes.title.value).toBe(props.title);
  expect(container.querySelector("img").attributes.alt.value).toBe(props.title);
  expect(container.querySelector(".card-title").innerHTML).toBe(props.title);
  expect(container.querySelector(".card-price").innerHTML).toBe(props.price);

  expect(container.firstChild).toMatchSnapshot();
});
