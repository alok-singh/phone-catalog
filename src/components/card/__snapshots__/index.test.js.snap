// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<a
  class="item-card"
  href="/link"
>
  <img
    alt="title"
    loading="lazy"
    src="image"
    title="title"
  />
  <div
    class="card-title"
  >
    title
  </div>
  <div
    class="card-price"
  >
    price
  </div>
</a>
`;
